src/foam/nanos/servlet/VirtualHostRoutingServlet.js

```diff
@@ -42,34 +74,180 @@ foam.CLASS({
     {
       name: 'init',
       type: 'Void',
-      args: [ { name: 'config', javaType: 'javax.servlet.ServletConfig' } ],
+      args: [ { name: 'config', javaType: 'ServletConfig' } ],
       javaCode: 'setServletConfig(config);',
       code: function() { }
     },
+    {
+      name: 'getTagString',
+      type: 'String',
+      documentation: `Utility method that returns an html tag string.`,
+      args: [ 
+        { name: 'tag', javaType: 'String'},
+        { name: 'keywords', javaType: 'ArrayList<String>'},
+        { name: 'attributes', javaType: 'HashMap'},
+        { name: 'hasClosingTag', javaType: 'Boolean'},
+        { name: 'isSingleClosingTag', javaType: 'Boolean'}
+      ],
+      javaCode: `
+        StringBuilder sb = new StringBuilder("<");
+        sb.append(tag);
+        sb.append(" ");
+        for ( String keyword : keywords ) {
+          sb.append(keyword);
+          sb.append(" ");
+        }
+        Iterator i = attributes.entrySet().iterator();
+        while ( i.hasNext() ) {
+          Entry attribute = (Entry) i.next();
+          sb.append(attribute.getKey());
+          sb.append("=\\"");
+          sb.append(attribute.getValue());
+          sb.append("\\" ");
+          i.remove();
+        }
+        if ( isSingleClosingTag ) {
+          sb.append("/>");
+          return sb.toString();
+        }
+        sb.append(">");
+        if ( hasClosingTag ) {
+          sb.append("</");
+          sb.append(tag);
+          sb.append(">");
+        }
+        return sb.toString();
+      `
+    },
+    {
+      name: 'populateHead',
+      type: 'Void',
+      documentation: `Generates the index file's head content based on theme and prints it to the response writer.`,
+      args: [ 
+        { name: 'theme', javaType: 'Theme'},
+        { name: 'out', javaType: 'PrintWriter'} 
+      ],
+      javaCode: `
+      ArrayList<HashMap> headConfig          = (ArrayList<HashMap>) theme.getHeadConfig();
+      Boolean            disableDefaultFonts = false;
+
+      out.println("<meta charset=\\"utf-8\\"/>");
+      out.println("<title>" + theme.getAppName() + "</title>");
```

Comment:
FYI: This is better is two print() statements and one final println() to avoid generating garbage.